### PR TITLE
add context to lambda var

### LINKED ISSUE #1893: [🐛 BUG] Lambda functions not working when page is imported
### What went wrong? 🤔

Lambda functions are not working anymore when the page is being imported into the main.py

Nothing appears on the page even if I ahve other visual elements in my page.

image

image

```python
[2024-10-03 11:47:17.965][Taipy][ERROR] Exception on /taipy-jsx/TaiPy_root_page [GET]
Traceback (most recent call last):
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\flask_cors\extension.py", line 194, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\gui.py", line 2409, in __render_page
    context = page.render(self)
              ^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\_page.py", line 38, in render
    self._rendered_jsx = self._renderer.render(gui)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\page.py", line 63, in render
    return self._base_element._render(gui)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\_element.py", line 183, in _render
    return f"{el[0]}{self._render_children(gui)}</{el[1]}>"
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\_element.py", line 186, in _render_children
    return "\n".join([child._render(gui) for child in self._children])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\_element.py", line 186, in <listcomp>
    return "\n".join([child._render(gui) for child in self._children])
                      ^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\_element.py", line 262, in _render
    self._evaluate_lambdas(gui)
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\builder\_element.py", line 73, in _evaluate_lambdas
    gui._bind_var_val(k, _getscopeattr(gui, expr))
                         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jacta\.conda\envs\develop\Lib\site-packages\taipy\gui\utils\_attributes.py", line 22, in _getscopeattr
    return getattr(gui._get_data_scope(), name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'types.SimpleNamespace' object has no attribute '__lambda_2569653502496'     
```

### Expected Behavior

It should work fine like before or like the "{}" syntax.

### Steps to Reproduce Issue

Create these files: main.py, page.py

main.py:

```python
from other_page import page
import taipy as tp

value = 10

tp.Gui(page).run()

```

other_page.py:

```python
import taipy.gui.builder as tgb

with tgb.Page() as page:
    tgb.text(lambda value: str(value))
```

Run the code.

### Version of Taipy

develop - 10/3/24

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1893

```py
import taipy as tp
from demo_1893_page_2 import page

value = 10

tp.Gui(page).run(title="1893 Lambda functions not working when page is imported", debug=True)

```

``` py title="demo_1893_page_2.py"
import taipy.gui.builder as tgb

with tgb.Page() as page:
    tgb.text(lambda value: str(value))

```

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]

**Review by namnguyen20999 — DISMISSED** [review]
Lgtm

File: taipy/gui/builder/_element.py [hunk 1]
```diff
@@ -67,6 +66,12 @@ def update(self, **kwargs):
         self._properties.update(kwargs)
         self.parse_properties()
 
+    @staticmethod
+    def __get_lambda_index():
+        _Element.__LAMBDA_VALUE_IDX += 1
+        _Element.__LAMBDA_VALUE_IDX %= 10000
```
Comment (FabienLelaquais) [inline taipy/gui/builder/_element.py]: Is this necessary?
If it's a fear for overflow, why not a far larger value? closer to MAX_INT?

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -25,6 +24,7 @@
 import warnings
 from importlib import metadata, util
 from importlib.util import find_spec
+from inspect import currentframe, getabsfile, ismethod, ismodule, isroutine
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py:27]: Yeah well... I don't really like those explicit imports because they hide the origin package, but hey... let's move on

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: taipy/gui/builder/_element.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/builder/_element.py]: I know ...
what do you think ?

File: taipy/gui/gui.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py:27]: easier to read

**Review by FabienLelaquais — APPROVED** [review]
👍

**Review by namnguyen20999 — APPROVED** [review]

**Pull request merged**